applications/app/controllers/OptInController.scala

```diff
@@ -6,16 +6,28 @@ import play.api.mvc.{Action, Controller, Cookie, DiscardingCookie}
 
 import scala.concurrent.duration._
 
-object OptInController extends Controller {
-  case class OptInFeature(cookieName: String, lifetime: Int = 90.days.toSeconds.toInt) {
-    def opt(choice: String) = choice match {
-      case "in" => optIn()
-      case _ => optOut()
-    }
-
-    def optIn() = SeeOther("/").withCookies(Cookie(cookieName, "true", maxAge = Some(lifetime)))
-    def optOut() = SeeOther("/").discardingCookies(DiscardingCookie(cookieName))
+trait OptInOutFeature extends Controller {
```

Comment:
Hmm. You're right but I was aiming to keep the semantics clear - you're opting _out_ of https, rather than opting _in_ to http. Could change it though if we feel it makes more sense.